Tests/SourceKitLSPTests/RenameTests.swift

```diff
@@ -0,0 +1,489 @@
+        _ = foo(b:)
+        """
+    )
+  }
+
+  func testRemoveLabelFromCompoundFunctionName() async throws {
+    try await assertRename(
+      """
+      func 1️⃣foo(a: Int) {}
+      _ = foo(a:)
+      """,
+      newName: "foo(_:)",
+      expected: """
+        func foo(_ a: Int) {}
+        _ = foo(_:)
+        """
+    )
+  }
+
+  func testIntroduceLabelToCompoundFunctionName() async throws {
+    try await assertRename(
+      """
+      func 1️⃣foo(_ a: Int) {}
+      _ = foo(_:)
+      """,
+      newName: "foo(a:)",
+      expected: """
+        func foo(a: Int) {}
+        _ = foo(a:)
+        """
+    )
+  }
+
+  func testRenameFromReference() async throws {
+    try await assertRename(
+      """
+      func foo(_ a: Int) {}
+      _ = 1️⃣foo(_:)
+      """,
+      newName: "foo(a:)",
+      expected: """
+        func foo(a: Int) {}
+        _ = foo(a:)
+        """
+    )
+  }
+
+  func testRenameMultipleParameters() async throws {
+    try await assertRename(
+      """
+      func 1️⃣foo(a: Int, b: Int) {}
+      foo(a: 1, b: 1)
+      """,
+      newName: "foo(x:y:)",
+      expected: """
+        func foo(x: Int, y: Int) {}
+        foo(x: 1, y: 1)
+        """
+    )
+  }
+
+  func testDontRenameParametersOmittedFromNewName() async throws {
+    try await assertRename(
+      """
+      func 1️⃣foo(a: Int, b: Int) {}
+      foo(a: 1, b: 1)
+      """,
+      newName: "foo(x:)",
+      expected: """
+        func foo(x: Int, b: Int) {}
+        foo(x: 1, b: 1)
+        """
+    )
+  }
+
+  func testIgnoreAdditionalParametersInNewName() async throws {
+    try await assertRename(
+      """
+      func 1️⃣foo(a: Int) {}
+      foo(a: 1)
+      """,
+      newName: "foo(x:y:)",
+      expected: """
+        func foo(x: Int) {}
+        foo(x: 1)
+        """
+    )
+  }
+
+  func testOnlySpecifyBaseNameWhenRenamingFunction() async throws {
+    try await assertRename(
+      """
+      func 1️⃣foo(a: Int) {}
+      foo(a: 1)
+      """,
+      newName: "bar",
+      expected: """
+        func bar(a: Int) {}
+        bar(a: 1)
+        """
+    )
+  }
+
+  func testIgnoreParametersInNewNameWhenRenamingVariable() async throws {
+    try await assertRename(
+      """
+      let 1️⃣foo = 1
+      _ = foo
+      """,
+      newName: "bar(x:y:)",
+      expected: """
+        let bar = 1
+        _ = bar
+        """
+    )
+  }
+
+  func testErrorIfNewNameDoesntContainClosingParenthesis() async throws {
+    // FIXME: syntactic rename does not support in-memory files... It should
+    let ws = try await IndexedSingleSwiftFileWorkspace(
+      """
+      func 1️⃣foo(a: Int) {}
+      foo(a: 1)
+      """
+    )
+    let request = RenameRequest(
+      textDocument: TextDocumentIdentifier(ws.fileURI),
+      position: ws.positions["1️⃣"],
+      newName: "bar(x:"
+    )
+    await assertThrowsError(try await ws.testClient.send(request))
+  }
+
+  func testErrorIfNewNameContainsTextAfterParenthesis() async throws {
```

Comment:
Same thing here, we ignore extra params, does extra text really matter?

Reply:
Addressing in/after the global rename PR.